Jellyfin.Api/Helpers/ItemHelper.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using Jellyfin.Api.Models.UserDtos;
+using LrcParser.Model;
+using LrcParser.Parser;
+using MediaBrowser.Controller.Entities;
+
+namespace Jellyfin.Api.Helpers
+{
+    /// <summary>
+    /// Item helper.
+    /// </summary>
+    public static class ItemHelper
+    {
+        /// <summary>
+        /// Opens lyrics file, converts to a List of Lyrics, and returns it.
+        /// </summary>
+        /// <param name="item">Requested Item.</param>
+        /// <returns>Collection of Lyrics.</returns>
+        internal static object? GetLyricData(BaseItem item)
+        {
+            // Find all classes that implement ILyricsProvider Interface
+            var foundLyricProviders = System.Reflection.Assembly.GetExecutingAssembly()
+                .GetTypes()
+                .Where(type => typeof(ILyricsProvider).IsAssignableFrom(type) && !type.IsInterface);
+
+            if (!foundLyricProviders.Any())
+            {
+                return null;
+            }
+
+            foreach (var provider in foundLyricProviders)
+            {
+                ILyricsProvider? newProvider = Activator.CreateInstance(provider) as ILyricsProvider;
```

Comment:
`ILyricsProvider` should be added to the DI pipeline.

Reply:
I've been poking around the DI pipeline, but have yet to really grasp the flow. Do we have any documentation, or an example PR, that I can look at and study?

Honestly, right now I don't even know where to begin.

Reply:
I would add `ILyricManager` and follow the `ISubtitleManager` flow as subtitles can be very similar to lyrics. I know this is getting much larger then the initial implementation but we would like to make the provider easily extendible by plugins

sample: Emby.Server.Implementations/ApplicationHost.cs L787

Reply:
Thanks man. I called in a friend who is a full time C# programmer to help me show me how to introduce ILyricsProvider into the DI pipeline. I have it working again, but we're adding the 2 providers individually to serviceCollection.

I suspect using an ILyricManager will allow only a single entry to serviceCollection to be used.